### PR TITLE
[ManagedServices] Update property descriptions before new API docs are published

### PR DESCRIPTION
**Pull request by skayani**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [X] Other, please clarify - Updating property descriptions before our new API version has public documentation published.
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [X] SDK of .NET (need service team to ensure code readiness)
      - [X] SDK of Python
      - [X] SDK of Java
      - [ ] SDK of Js
      - [x] SDK of Go
      - [X] PowerShell
      - [X] CLI
      - [X] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json [hunk 1]
```diff
@@ -574,65 +574,65 @@
...
         },
         "registrationDefinitionName": {
           "type": "string",
-          "description": "Name of the registration definition."
+          "description": "The name of the registration definition."
         },
         "managedByTenantId": {
           "type": "string",
-          "description": "Id of the managedBy tenant."
+          "description": "The ID of the managedBy tenant."
```
Comment (dramuy) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json:630]: >ID of the managedBy tenan [](http://example.com/codeflow?start=30&length=25)

The identifier for managedBy tenant.

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json [hunk 2]
```diff
@@ -1010,29 +1010,29 @@
...
       "type": "object",
-      "description": "List of the operations."
+      "description": "The list of the operations."
     },
     "Authorization": {
       "properties": {
         "principalId": {
           "type": "string",
-          "description": "Principal Id of the security group/service principal/user that would be assigned permissions to the projected subscription"
+          "description": "The identifier of the principal."
```
Comment (dramuy) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json:1023]: >The identifier of the principal. [](http://example.com/codeflow?start=26&length=32)

I am thinking should be mention about AAD principal or something to convey it is AAD objectId.

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json [hunk 3]
```diff
@@ -574,65 +574,65 @@
...
-          "description": "Description of the registration definition."
+          "description": "The description of the registration definition."
         },
         "authorizations": {
           "type": "array",
           "items": {
             "$ref": "#/definitions/Authorization"
           },
-          "description": "Authorization tuple containing principal id of the user/security group or service principal and id of the build-in role."
+          "description": "The collection of principals in the managedBy tenant with permanent access to the managed tenant."
```
Comment (dramuy) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json:615]: >The collection of principals in the managedBy tenant with permanent access to the managed tenant [](http://example.com/codeflow?start=26&length=96)

The collection of principals in the managedBy tenant with permanent access during the delegation on the delegated resource.

We should convey that permanent access is only during the delegation period and also not to the whole managed tenant.

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json [hunk 4]
```diff
@@ -745,29 +745,29 @@
...
-                  "description": "Description of the registration definition."
+                  "description": "The description of the registration definition."
                 },
                 "authorizations": {
                   "type": "array",
                   "items": {
                     "$ref": "#/definitions/Authorization"
                   },
-                  "description": "Authorization tuple containing principal id of the user/security group or service principal and id of the build-in role."
+                  "description": "The collection of principals in the managedBy tenant with permanent access to the managed tenant."
```
Comment (dramuy) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json:755]: >The collection of principals in the managedBy tenant with permanent access to the managed tenant. [](http://example.com/codeflow?start=34&length=97)

this need to be fixed as per other comment.

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json [hunk 5]
```diff
@@ -745,29 +745,29 @@
...
                 },
                 "eligibleAuthorizations": {
                   "type": "array",
                   "items": {
                     "$ref": "#/definitions/EligibleAuthorization"
                   },
-                  "description": "Eligible PIM authorization tuple containing principal id of the user/security group or service principal, id of the built-in role, and just-in-time access policy setting"
+                  "description": "The user ID, role, and just-in-time access policy for a principal in the managedBy tenant."
```
Comment (dramuy) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json:762]: >The user ID, role, and just-in-time access policy for a principal in the managedBy tenant." [](http://example.com/codeflow?start=34&length=91)

why eligible authorizations is having different approach of description than authorizations.

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json [hunk 6]
```diff
@@ -887,21 +887,21 @@
...
-          "description": "Id of the managedBy tenant."
+          "description": "The identifier of the managedBy tenant."
         },
         "authorizations": {
           "type": "array",
           "items": {
             "$ref": "#/definitions/Authorization"
           },
-          "description": "Authorization tuple containing principal id of the user/security group or service principal and id of the build-in role."
+          "description": "The collection of principals in the managedBy tenant with permanent access to the managed tenant."
```
Comment (dramuy) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json:897]: >The collection of principals in the managedBy tenant with permanent access to the managed tenant. [](http://example.com/codeflow?start=26&length=97)

address it as per previous comment.

Comment (dramuy) [reply specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json:897]: also these are not only just collection of principles.

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json [hunk 7]
```diff
@@ -887,21 +887,21 @@
...
         },
         "eligibleAuthorizations": {
           "type": "array",
           "items": {
             "$ref": "#/definitions/EligibleAuthorization"
           },
-          "description": "Eligible PIM authorization tuple containing principal id of the user/security group or service principal, id of the built-in role, and just-in-time access policy setting"
+          "description": "The user ID, role, and just-in-time access policy for a principal in the managedBy tenant."
```
Comment (dramuy) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json:904]: >The user ID, role, and just-in-time access policy for a principal in the managedBy tenant [](http://example.com/codeflow?start=26&length=89)

revisit based on authorizations description.

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json [hunk 8]
```diff
@@ -1044,24 +1044,24 @@
         "roleDefinitionId"
       ],
       "type": "object",
-      "description": "Authorization tuple containing principal Id (of user/service principal/security group) and role definition id."
+      "description": "The user ID and permanent role for a principal in the managedBy tenant."
```
Comment (dramuy) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json:1047]: >permanent role [](http://example.com/codeflow?start=38&length=14)

there is no concept like "permanent role"

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json [hunk 9]
```diff
@@ -1010,29 +1010,29 @@
...
       "properties": {
         "principalId": {
           "type": "string",
-          "description": "Principal Id of the security group/service principal/user that would be assigned permissions to the projected subscription"
+          "description": "The identifier of the Azure Active Directory principal."
         },
         "principalIdDisplayName": {
           "type": "string",
-          "description": "Display name of the principal Id."
+          "description": "The display name of the principal."
```
Comment (dramuy) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json:1027]: >principal [](http://example.com/codeflow?start=50&length=9)

The display name of the Azure Active Directory principal.

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json [hunk 10]
```diff
@@ -1044,24 +1044,24 @@
...
       "properties": {
         "principalId": {
           "type": "string",
-          "description": "Principal Id of the security group/service principal/user that would be delegated permissions to the projected subscription"
+          "description": "The identifier of the Azure Active Directory principal."
         },
         "principalIdDisplayName": {
           "type": "string",
-          "description": "Display name of the principal Id."
+          "description": "The display name of the principal."
```
Comment (dramuy) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json:1057]: >principal [](http://example.com/codeflow?start=50&length=9)

AAD principal

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json [hunk 11]
```diff
@@ -1044,24 +1044,24 @@
...
         },
         "roleDefinitionId": {
           "type": "string",
-          "description": "The role definition identifier. This role will delegate all the permissions that the security group/service principal/user must have on the projected subscription. This role cannot be an owner role."
+          "description": "The identifier of the Azure built-in role that defines the permissions that the principal will have on the projected scope."
```
Comment (dramuy) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json:1061]: >principal [](http://example.com/codeflow?start=106&length=9)

AAD principal

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json [hunk 12]
```diff
@@ -1010,29 +1010,29 @@
...
         },
         "principalIdDisplayName": {
           "type": "string",
-          "description": "Display name of the principal Id."
+          "description": "The display name of the principal."
         },
         "roleDefinitionId": {
           "type": "string",
-          "description": "The role definition identifier. This role will define all the permissions that the security group/service principal/user must have on the projected subscription. This role cannot be an owner role."
+          "description": "The identifier of the Azure built-in role that defines the permissions that the principal will have on the projected scope."
```
Comment (dramuy) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json:1031]: >principal [](http://example.com/codeflow?start=106&length=9)

AAD principal

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json [hunk 13]
```diff
@@ -1070,30 +1070,30 @@
...
       "properties": {
         "principalId": {
           "type": "string",
-          "description": "Principal Id of the user or security group that will approve JIT activation requests for the eligible authorization."
+          "description": "The identifier of the Azure Active Directory principal."
         },
         "principalIdDisplayName": {
           "type": "string",
-          "description": "Display name of the principal Id."
+          "description": "The display name of the principal."
```
Comment (dramuy) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json:1083]: >principal [](http://example.com/codeflow?start=50&length=9)

AAD principal

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json [hunk 14]
```diff
@@ -1070,30 +1070,30 @@
...
         }
       },
       "required": [
         "principalId"
       ],
       "type": "object",
-      "description": "A principal Id and user-friendly display name representing an eligible authorization approver."
+      "description": "Defines the principal that will need to approve any just-in-time access requests by the principal defined in the EligibleAuthorization."
```
Comment (dramuy) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json:1090]: >principal [](http://example.com/codeflow?start=34&length=9)

defines the list of AAD principals that can* approve

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json (hunk 3 above)
Comment (skayani) [reply specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json]: Updated the wording a bit to convey the message

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json (hunk 6 above)
Comment (skayani) [reply specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json]: Should be fixed now w/ new wording

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json (hunk 8 above)
Comment (skayani) [reply specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json]: Good point - changed

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json (hunk 10 above)
Comment (skayani) [reply specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json]: Changed most references of "principal" to "Azure Active Directory principal"

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json (hunk 7 above)
Comment (dramuy) [reply specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json]: describing "the" just-in-time access..

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json [hunk 15]
```diff
@@ -887,21 +887,21 @@
...
-          "description": "Id of the managedBy tenant."
+          "description": "The identifier of the managedBy tenant."
         },
         "authorizations": {
           "type": "array",
           "items": {
             "$ref": "#/definitions/Authorization"
           },
-          "description": "Authorization tuple containing principal id of the user/security group or service principal and id of the build-in role."
+          "description": "The collection of authorization objects describing what access Azure Active Directory principals in the managedBy tenant will receive on the delegated resource in the managed tenant."
```
Comment (dramuy) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json:897]: >what [](http://example.com/codeflow?start=77&length=4)

describing "the" access...

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json [hunk 16]
```diff
@@ -745,29 +745,29 @@
...
-                  "description": "Authorization tuple containing principal id of the user/security group or service principal and id of the build-in role."
+                  "description": "The collection of authorization objects describing what access Azure Active Directory principals in the managedBy tenant will receive on the delegated resource in the managed tenant."
                 },
                 "eligibleAuthorizations": {
                   "type": "array",
                   "items": {
                     "$ref": "#/definitions/EligibleAuthorization"
                   },
-                  "description": "Eligible PIM authorization tuple containing principal id of the user/security group or service principal, id of the built-in role, and just-in-time access policy setting"
+                  "description": "The collection of eligible authorization objects describing what just-in-time access Azure Active Directory principals in the managedBy tenant will receive on the delegated resource in the managed tenant."
```
Comment (dramuy) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json:762]: >what [](http://example.com/codeflow?start=94&length=4)

describing "the" just-in-time

**Review by dramuy — APPROVED** [review]
:shipit:

**Comment by skayani** [comment]
Hi @jianyexi , can we have this change merged please?

**Pull request merged**